tests/unit_tests/data_steward/cdr_cleaner/reporter_test.py

```diff
@@ -0,0 +1,240 @@
+        }, {
+            'name': 'bar',
+        }]
+        self.assertEqual(actual, expected)
+
+    def test_format_values_with_sql_field(self):
+        # preconditions
+        report_dicts = [{'sql': [{'query': 'query one'}]}]
+        fields_list = ['sql']
+
+        # test
+        with self.assertLogs('cdr_cleaner.reporter', level='DEBUG') as cm:
+            actual = reporter.format_values(report_dicts, fields_list)
+
+        expected_logs = []
+        log_record = cm.records[0]
+        self.assertEqual(log_record.levelno, logging.DEBUG)
+        self.assertEqual(log_record.msg, 'SQL field exists')
+
+        expected = [{'sql': 'query one'}]
+        self.assertEqual(actual, expected)
+
+    def test_format_values_raises_errors(self):
+        # preconditions
+        report_dicts = [{'name': 'foo', 'jira-issues': [0, 1]}]
+        fields_list = ['name', 'jira-issues']
+
+        # test
+        with self.assertLogs('', level='INFO') as cm:
+            self.assertRaises(TypeError, reporter.format_values, report_dicts,
+                              fields_list)
+
+        expected_logs = []
+        log_record = cm.records[0]
+        self.assertEqual(log_record.levelno, logging.ERROR)
+        self.assertEqual(
+            log_record.msg,
+            'erroneous field is jira-issues\nerroneous value is [0, 1]')
+
+    def test_format_values(self):
+        # preconditions
+        report_dicts = [{'name': 'foo', 'jira-issues': ['DC-000', 'DC-0']}]
+        fields_list = ['name', 'jira-issues']
+
+        # test
+        actual = reporter.format_values(report_dicts, fields_list)
+
+        # post conditions
+        expected = [{'name': 'foo', 'jira-issues': 'DC-000, DC-0'}]
+        self.assertEqual(actual, expected)
+
+    def test_parse_args(self):
+        # preconditions
+        arg_list = [
+            '--data-stage', 'rdr', 'ehr', '--fields', 'name', 'sandbox-tables',
+            'description', '--output-file', 'temp.csv'
+        ]
+
+        # test
+        elements = reporter.parse_args(arg_list)
+
+        #post conditions
+        self.assertEqual(elements.data_stage, ['rdr', 'ehr'])
+        self.assertEqual(elements.fields,
+                         ['name', 'sandbox-tables', 'description'])
+        self.assertEqual(elements.output_filepath, 'temp.csv')
+        self.assertFalse(elements.console_log)
+
+    def test_get_function_info(self):
+        # preconditions
+        fields_list = ['name', 'module', 'description', 'sql', 'depends-on']
+
+        # test
+        actual = reporter.get_function_info(mock_function_obj, fields_list)
+
+        # post condition
+        expected = {
+            'name': mock_function_obj.__name__,
+            'module': mock_function_obj.__module__,
+            'description': mock_function_obj.__doc__,
+            'sql': 'unknown',
+            'depends-on': 'unknown',
+        }
+
+        self.assertEqual(actual, expected)
+
+    def test_get_stage_elements(self):
```

Comment:
This test passes, but I receive the following `AttributeError` when running the unit tests locally:

```Traceback (most recent call last):
  File "/Users/aimeemcgrenera/Documents/dev/curation/data_steward/cdr_cleaner/reporter.py", line 108, in get_stage_elements
    _ = instance.get_query_specs()
AttributeError: 'list' object has no attribute 'get_query_specs'```

It seems to me you are just assigning the instance to `_`, but that variable is irrelevant?

Reply:
I'm glad you noticed.  This is testing a few things.  

1. The lists of query cleaning classes and functions each require different handling.  So line 108 in reporter helps identify a class vs a function.  The error is caught and the function information that can be found is stored.  But functions don't have all the same data as classes, so they have to be handled differently.  This is an example of using a try/catch to differentiate different types instead of using an `isinstance` specific check.  the exception is caught and handled on line 138.
2.  This is making sure that everything in the lists is handled one way or another.  If a function sneaks into the list with a matching `get_query_specs` signature, this test will likely fail.